source/gpu_perf_api_common/gpa_context.cc

```diff
@@ -419,7 +419,7 @@ bool GpaContext::IsAmdDevice() const
     GpaUInt32 vendor_id;
     bool      is_amd = false;
 
-    if (hw_info_.GetVendorId(vendor_id) && kAmdVendorId == vendor_id)
+    if (hw_info_.GetVendorId(vendor_id) && ((kAmdVendorId == vendor_id) || (kSamsungVendorId == vendor_id)))
     {
         is_amd = true;
     }
```

Comment:
I'd prefer to keep this function intuitive and only return true if it is truly an AMD device, just in case we do need to differentiate between the two.

Can you please add a separate IsSamsungDevice()? and optionally an IsAmdOrSamsungDevice()? Refer to them as appropriate in other parts of the code.